### PR TITLE
Docker: Sanitize android docker

### PR DESCRIPTION
**Pull request by heliocastro**

### CODE REVIEW

**Review by tsteenbe — APPROVED** [review]

**Review by tsteenbe — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -241,7 +241,19 @@ RUN curl -sSL https://get.haskellstack.org/ | bash -s -- -d /usr/bin
...
+FROM ort-base-image AS androidbuild
+
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    sudo apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive sudo apt-get install -y --no-install-recommends \
+    unzip \
+    && sudo rm -rf /var/lib/apt/lists/*
+
+RUN curl -ksS https://storage.googleapis.com/git-repo-downloads/repo | sudo tee /usr/bin/repo > /dev/null 2>&1 \
```
Comment (fviernau) [inline Dockerfile]: Why is  Git-Repo now installed in `androidbuild`?  (I know that Git-Repo is used for development of AOSP, but is this the reason to install it in the `androidBuild`?

File: Dockerfile [hunk 2]
```diff
@@ -267,8 +267,6 @@ RUN --mount=type=tmpfs,target=/android \
     fi \
     && yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS --sdk_root=$ANDROID_HOME "platform-tools" "cmdline-tools;latest"
 
-COPY docker/android.sh /etc/profile.d
```
Comment (fviernau) [inline Dockerfile:263]: This becaume unneeded when `android` image / layer was taken into use? Would it make sense to squash this as well?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (heliocastro) [reply Dockerfile:263]: i do prefer keep in a separate commit. Was squashed before, and i was told to create separate commits

File: Dockerfile [hunk 3]
```diff
@@ -338,9 +338,9 @@ COPY --from=haskellbuild /usr/bin/stack /usr/bin
 
 # Repo and Android
 ENV ANDROID_HOME=/opt/android-sdk
-COPY --from=androidbuild /usr/bin/repo /usr/bin/
-COPY --from=androidbuild /etc/profile.d/android.sh /etc/profile.d/
-COPY --chown=$USERNAME:$USERNAME --from=androidbuild ${ANDROID_HOME} ${ANDROID_HOME}
+ENV PATH=$PATH:${ANDROID_HOME}/tools/bin:${ANDROID_HOME}/emulator:${ANDROID_HOME}/platform-tools
+COPY --from=android /usr/bin/repo /usr/bin/
+COPY --from=android ${ANDROID_HOME} ${ANDROID_HOME}
```
[the remark this answers is not in the document]
Comment (heliocastro) [reply Dockerfile:341]: i do prefer keep in a separate commit. Was squashed before, and i was told to create separate commits

File: Dockerfile [hunk 4]
```diff
@@ -338,9 +338,9 @@ COPY --from=haskellbuild /usr/bin/stack /usr/bin
 
 # Repo and Android
 ENV ANDROID_HOME=/opt/android-sdk
-COPY --from=androidbuild /usr/bin/repo /usr/bin/
-COPY --from=androidbuild /etc/profile.d/android.sh /etc/profile.d/
-COPY --chown=$USERNAME:$USERNAME --from=androidbuild ${ANDROID_HOME} ${ANDROID_HOME}
+ENV PATH=$PATH:${ANDROID_HOME}/tools/bin:${ANDROID_HOME}/emulator:${ANDROID_HOME}/platform-tools
```
[the remark this answers is not in the document]
Comment (heliocastro) [reply Dockerfile:339]: Normally yes when you have projects that assign different architectures, but then, i can't judge the relevancy on Ort and if any project bootstrap for scanning will need it.
Do you think i should remove ?

File: Dockerfile (hunk 1 above)
Comment (heliocastro) [reply Dockerfile]: Good catch. It's a bug. Fixing

File: Dockerfile (hunk 2 above)
Comment (fviernau) [reply Dockerfile:263]: Ok, I didn't review previous discussions and don't want to introduce back and force.

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply Dockerfile]: no strong opinion on that one. Slight tendency to remove, as I expect it's related to "using" the emulator. Anyhow, up to you.

File: Dockerfile (hunk 1 above)
Comment (fviernau) [reply Dockerfile]: Is it supposed to be fixed already or will you force push again?

Comment (tsteenbe) [reply Dockerfile]: Was fixed.

**Review by fviernau — APPROVED** [review]

**Pull request merged**